tests/conftest.py

```diff
@@ -19,6 +29,167 @@
+        yield
+    else:
+        engine = sqlalchemy.create_engine(f"sqlite:///{DB_NAME}", future=True)
+
+        # get the temp directory shared by all workers
+        root_tmp_dir = tmp_path_factory.getbasetemp().parent
+        lock = root_tmp_dir / (DB_NAME + ".lock")
+
+        # Create the db if it does not exist already.
+        with filelock.FileLock(lock):
+            if not os.path.exists(DB_NAME):
+                with engine.connect() as conn:
+                    conn.execute(sqlalchemy.text("VACUUM"))
+
+        # TODO use alembic for database migration like Coiled Benchmarks does
+
+        yield engine
+
+
+@pytest.fixture(scope="function")
+def benchmark_db_session(benchmark_db_engine):
+    """SQLAlchemy session for a given test.
+
+    Most tests shouldn't need to include this fixture directly.
+
+    Yields
+    ------
+    SQLAlchemy session for the benchmark database engine if run as part of a benchmark,
+    None otherwise.
+    """
+    if not benchmark_db_engine:
+        yield
+    else:
+        with Session(benchmark_db_engine, future=True) as session, session.begin():
+            yield session
+
+
+@pytest.fixture(scope="function")
+def test_run_benchmark(benchmark_db_session, request, testrun_uid):
+    """SQLAlchemy ORM object representing a given test run.
+
+    By including this fixture in a test (or another fixture that includes it)
+    you trigger the test being written to the benchmark database. This fixture
+    includes data common to all tests, including python version, test name,
+    and the test outcome.
+
+    Yields
+    ------
+    SQLAlchemy ORM object representing a given test run if run as part of a benchmark,
+    None otherwise.
+    """
+    if not benchmark_db_session:
+        yield
+    else:
+        run = TestRun(
+            session_id=testrun_uid,
+            name=request.node.name,
+            originalname=request.node.originalname,
+            path=str(request.node.path.relative_to(TEST_DIR)),
+            cubed_version=cubed.__version__,
+            #cubed_xarray_version=cubed_xarray.__version__,
+            #xarray_version=xarray.__version__,
+            #lithops_version=lithops.__version__, 
+            python_version=".".join(map(str, sys.version_info)),
+            platform=sys.platform,
+            ci_run_url=WORKFLOW_URL,
+        )
+        yield run
+
+        rep = getattr(request.node, "rep_setup", None)
+        if rep:
+            run.setup_outcome = rep.outcome
+        rep = getattr(request.node, "rep_call", None)
+        if rep:
+            run.call_outcome = rep.outcome
+        rep = getattr(request.node, "rep_teardown", None)
+        if rep:
+            run.teardown_outcome = rep.outcome
+
+        benchmark_db_session.add(run)
+        benchmark_db_session.commit()
+
+
+@pytest.fixture(scope="function")
+def benchmark_time(test_run_benchmark):
+    """
+    Benchmark the wall clock time of executing some code.
```

Comment:
This is just an MVP of the simplest thing we would want to benchmark. Ultimately we want a general `benchmark` fixture which calls `benchmark_time`, `benchmark_memory` etc.